app/jobs/clock/refresh_draft_invoices_job.rb

```diff
@@ -6,6 +6,8 @@ class RefreshDraftInvoicesJob < ApplicationJob
 
     def perform
       Invoice.ready_to_be_refreshed.find_each do |invoice|
```

Comment:
Maybe we should check the condition directly in a DB query. Since we could have a lot of invoice to refresh, we might want to avoid performing one query per invoice. What do you think?

Reply:
Makes sense! Updated